procrastinate/utils.py

```diff
@@ -235,8 +235,12 @@ def _get_module_name(obj: Any) -> str:
 
 
 def get_full_path(obj: Any) -> str:
-
-    return f"{_get_module_name(obj)}.{obj.__name__}"
+    try:
+        return f"{_get_module_name(obj)}.{obj.__name__}"
+    except AttributeError as exc:
```

Comment:
Hm, I wonder if it's very obvious that _get_module_name() may raise an AttributeError.

- Either that should be a custom ProcrastinateException but that might (or might not) be overkill
- or we could add a line in that function's docstring saying it may raise an AttributeError and in what case.

Reply:
the AttributeError isn't coming from _get_module, it's coming from `obj.__name__` in this function

(I may be misunderstanding your question here)

Reply:
Ah damn you're right, I read too fast. Then, can we change the code so that we do `obj.__name__` alone it its try/except clause and only then we return the f-string ?

Reply:
yup! fixed